### PR TITLE
[Data Object] Table Data-Type should always return an array

### LINKED ISSUE #6661: [Data Object] Table Data-Type should always return an array
At the moment the data-type returns and empty string or even `null` if no value is set. 
With v7 it should always return an array, in case of no data, just an empty array (`[]`). 

See original discussion here: #6532

### PR DESCRIPTION
**Pull request by weisswurstkanone**
fixes #6661

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/Table.php [hunk 1]
```diff
@@ -326,7 +331,7 @@ public function getDataFromResource($data, $object = null, $params = [])
         $unserializedData = Serialize::unserialize((string) $data);
 
         if ($data === null || $unserializedData === null) {
-            return null;
+            return [];
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/Table.php:334]: Change also PHPDoc:
`@return array`

Comment (weisswurstkanone) [reply models/DataObject/ClassDefinition/Data/Table.php:334]: done

File: models/DataObject/ClassDefinition/Data/Table.php [hunk 2]
```diff
@@ -100,7 +100,7 @@ class Table extends Data implements ResourcePersistenceAwareInterface, QueryReso
      *
      * @var string
      */
-    public $phpdocType = 'array|string';
+    public $phpdocType = 'array';
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Table.php:103]: we should rebuild classes to update phpdoc

Comment (dvesh3) [reply models/DataObject/ClassDefinition/Data/Table.php:103]: nevermind! It should be covered in a generic task for other changes on v7.

**Pull request merged**